### PR TITLE
Activate gettext translation at the starting of calaos_server...

### PR DESCRIPTION
**Pull request by FragJage**
… to have --help and --gendoc in the local language.
I'm not sure for the path on bindtextdomain function...

### CODE REVIEW

**Comment by raoulh** [comment]
I would use PACKAGE_DATA_DIR instead of hard coded /usr/... which is wrong.

Also check the log in travis, the build is failing for macos

**Comment by FragJage** [comment]
PACKAGE_DATA_DIR is set to /usr/local/share/calaos but make install copy translation files (*.po) to /usr/local/share/locale. It's a bug ?

**Comment by naguirre** [comment]
You should have a PACKAGE_LOCAL_DIR, if it's not the case we need to add it
in Makefile.am.

Le 4 oct. 2017 05:56, "François JARRIER-GELLEZ" <notifications@github.com>
a écrit :

PACKAGE_DATA_DIR is set to /usr/local/share/calaos but make install copy
translation files (*.po) to /usr/local/share/locale. It's a bug ?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/calaos/calaos_base/pull/140#issuecomment-333990946>,
or mute
the thread

File: scripts/osx/install.sh [hunk 1]
```diff
@@ -7,7 +7,8 @@ source $SCRIPTDIR/../lib.sh
 
 #install dependencies
 brew update
-brew install libtool gettext libuv libsigc++ jansson curl luajit owfs libusb ola
+brew install libtool gettext libuv libsigc++ jansson curl luajit owfs libusb ola gettext
```
Comment (raoulh) [inline scripts/osx/install.sh]: gettext was already in the list of package

**Comment by raoulh** [comment]
About the failing build on macOS:
https://www.gnu.org/software/gettext/FAQ.html#integrating_undefined

**Comment by FragJage** [comment]
Very good FAQ. Thanks.

**Comment by FragJage** [comment]
With LibIntl, executables are built but gmo/mo files build fail. It's because sed in OS X is a little bit different. I have found that some project disable NLS to build on osx.

**Comment by raoulh** [comment]
What are the side effects of disabling nls?

**Comment by FragJage** [comment]
NLS is Native Language Support. gettext is disable. This was already the case on your travis build 23.3 : see https://travis-ci.org/calaos/calaos_base/jobs/281186799 lines 2062 to 2066, library for gettext was not found. I added "brew link --force gettext" for the library could be found.

**Comment by raoulh** [comment]
Hmm, you are right. Actually the gettext package was required for running autogen.sh and glibtoolize.

I will do some research on my mac for building calaos_server with NLS enabled.

**Comment by FragJage** [comment]
I see two ways :
- display po file to see the problem (i can't do that with travis)
- try to disable mo files build (i believe there are identical between Mac and Linux)

**Comment by raoulh** [comment]
Can be merged. Thanks a lot!

**Pull request merged**